CMakeLists.txt

```diff
@@ -6,8 +6,9 @@ if(NOT CMAKE_C_STANDARD)
 endif()
 
 if(NOT CMAKE_CXX_STANDARD)
-  set(CMAKE_CXX_STANDARD 14)
+  set(CMAKE_CXX_STANDARD 17)
 endif()
+set( CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -fPIC" )
```

Comment:
Don't set this globally. Instead, set it one by one on each target that needed it, e.g.:
```cmake
add_library(${PROJECT_NAME} SHARED
  ...
)

target_compile_options(${PROJECT_NAME} PRIVATE -fPIC)
```